src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -157,39 +159,50 @@ class GrpcDriver(Driver):
         The version of the FAB used in the run.
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
-        driver_service_address: str = DEFAULT_SERVER_ADDRESS_DRIVER,
-        root_certificates: Optional[bytes] = None,
-        fab_id: Optional[str] = None,
-        fab_version: Optional[str] = None,
+        run_id: int,
+        stub: Optional[GrpcDriverStub] = None,
     ) -> None:
-        self.addr = driver_service_address
-        self.root_certificates = root_certificates
-        self.driver_helper: Optional[GrpcDriverHelper] = None
-        self.run_id: Optional[int] = None
-        self.fab_id = fab_id if fab_id is not None else ""
-        self.fab_version = fab_version if fab_version is not None else ""
+        self.stub = stub
+        self._run_id = run_id
+        self._fab_id = ""
+        self._fab_ver = ""
```

Comment:
It'd be better to change the type to `Optional[str]` here instead of using a "magic" empty value

Reply:
Cool! Thanks for the suggestion.